src/Mono.Android/Android.Runtime/AndroidEnvironment.cs

```diff
@@ -254,8 +294,12 @@ static string GetDefaultTimeZone ()
 		static SynchronizationContext GetDefaultSyncContext ()
 		{
 			var looper = Android.OS.Looper.MainLooper;
+			try {
 			if (Android.OS.Looper.MyLooper() == looper)
 				return Android.App.Application.SynchronizationContext;
+			} catch (Java.Lang.Throwable ex) when (JNIEnv.ShouldWrapJavaException (ex)) {
```

Comment:
I suspect we should remove the `when` here, and just catch `System.Exception`.  There is likely *no sane scenario* in which `SynchronizationContext.Current` should throw an exception.